client/metrics/metrics.go

```diff
@@ -26,6 +27,26 @@ var initialized int32
 func init() {
 	initMetrics(prometheus.Labels{})
 	initCmdDurations()
+	initRegisteredConsumers()
+}
+
+var mutex sync.Mutex
+var consumersInitializers []func()
```

Comment:
What about using an `atomic.Value` to store `[]func()`?

Reply:
Your consideration is thorough. My initial intention was to merge the lock on line 33 with line 34 into one structure, making it look more elegant.

Reply:
I still prefer using a single structure, you could use a customed structure rather than `atomic.Value`.